internal/services/loadbalancer/loadbalancer_nat_rule_resource_test.go

```diff
@@ -47,6 +47,36 @@ func TestAccAzureRMLoadBalancerNatRule_complete(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMLoadBalancerNatRule_mapToBackendAddressPool(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_lb_nat_rule", "test")
+	r := LoadBalancerNatRule{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.mapToBackendAddressPool(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccAzureRMLoadBalancerNatRule_multipleRuleMapToBackendAddressPool(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_lb_nat_rule", "test")
+	r := LoadBalancerNatRule{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.multipleRuleMapToBackendAddressPool(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
```

Comment:
Can we add another test for update from using the BAP to using the backend_port, and vice versa?

Reply:
> Can we add another test for update from using the BAP to using the backend_port, and vice versa?
do you mean to update the backend port ?

Reply:
I added the test case from regular to nat rule v2 (using the port range and the backend address pool id), however, the operation cannot be done from the other way around. service confirmed this bug and they are working on the fix now.